atlasaction/action.go

```diff
@@ -51,11 +50,16 @@ type (
 		// SetOutput sets the value of the output with the given name.
 		SetOutput(string, string)
 		// GetTriggerContext returns the context of the trigger event.
-		GetTriggerContext() (*TriggerContext, error)
-		// AddStepSummary adds a summary to the action step.
-		AddStepSummary(string)
+		GetTriggerContext(context.Context) (*TriggerContext, error)
 	}
 
+	// Reporter is an interface for reporting the status of the actions.
+	Reporter interface {
+		MigrateApply(context.Context, *atlasexec.MigrateApply)
+		MigrateLint(context.Context, *atlasexec.SummaryReport)
+		SchemaPlan(context.Context, *atlasexec.SchemaPlan)
+		SchemaApply(context.Context, *atlasexec.SchemaApply)
```

Comment:
May be just one method?
```suggestion
		Report(context.Context, any)
```

Reply:
Why? Do you prefer type-switch over separate functions? I don't, because this way the callee don't need to handle unexpected inputs.